src/main/java/olivero/parsers/TaskParser.java

```diff
@@ -0,0 +1,92 @@
+package olivero.parsers;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+import olivero.exceptions.TaskParseException;
+import olivero.tasks.Deadline;
+import olivero.tasks.Event;
+import olivero.tasks.TaskList;
+import olivero.tasks.ToDo;
+
+/**
+ * Parses raw tasks into a {@code TaskList} object.
+ */
+public class TaskParser {
+
+    /** Argument separator used in the raw data formatting to demarcate arguments of each task. */
+    public static final String SEPARATOR = " | ";
+
+    /**
+     * Returns a {@code TaskList} object that has been parsed from the raw string input.
+     *
+     * @param rawTasks The raw string input containing the list of tasks.
+     * @return Parsed {@code TaskList} object.
+     * @throws TaskParseException If the provided format of the raw tasks is invalid during parsing.
+     */
+    public TaskList parse(String rawTasks) throws TaskParseException {
+        TaskList taskList = new TaskList();
+        if (rawTasks.isEmpty()) {
+            return taskList;
+        }
+        String[] lines = rawTasks.split(System.lineSeparator());
+        for (String line : lines) {
+            String[] args = line.split(Pattern.quote(SEPARATOR));
```

Comment:
```suggestion
            String[] args = line.split(Pattern.quote(SEPARATOR), limit);
```

Consider taking into account cases where the task description has '|' with split()'s limit argument